internal/cmd/request.go

```diff
@@ -105,13 +107,107 @@ func PrintCMDResponseFull(request any, response core.APIResponse, responseType a
 		fmt.Printf("\nAutograder Response:\n---\n%s\n---\n", util.MustToJSONIndent(response))
 	}
 
+	successfulConversion := false
+	customPrintOutput := ""
 	if customPrintFunc != nil {
-		fmt.Println(customPrintFunc(response))
-	} else if responseType == nil {
-		fmt.Println(util.MustToJSONIndent(response.Content))
+		customPrintOutput, successfulConversion = customPrintFunc(response)
+	}
+
+	output := ""
+	if successfulConversion {
+		output = customPrintOutput
 	} else {
-		responseContent := reflect.New(reflect.TypeOf(responseType)).Interface()
-		util.MustJSONFromString(util.MustToJSON(response.Content), &responseContent)
-		fmt.Println(util.MustToJSONIndent(responseContent))
+		if responseType != nil {
+			responseContent := reflect.New(reflect.TypeOf(responseType)).Interface()
+			util.MustJSONFromString(util.MustToJSON(response.Content), &responseContent)
+			output = util.MustToJSONIndent(responseContent)
+		} else {
+			output = util.MustToJSONIndent(response.Content)
+		}
+	}
+
+	fmt.Println(output)
+}
+
+// Attempt to convert an API response into a TSV table.
+// If the response content has multiple keys or a single map, it creates a single row table.
+// If it has one key with a list, the values in each element becomes a row of the table.
+// Return the table and true on a successful conversion or false if there is an issue converting.
+func ConvertApiResponseToTable(response core.APIResponse) (string, bool) {
+	responseContent, ok := response.Content.(map[string]any)
+	if !ok {
+		return "", false
+	}
+
+	if len(responseContent) == 0 {
+		return "", false
+	} else if len(responseContent) > 1 {
+		return convertEntriesToTable([]any{responseContent})
+	} else {
+		var responseContentKey string
+		for key := range responseContent {
+			responseContentKey = key
+		}
+
+		var entries []any
+		switch value := responseContent[responseContentKey].(type) {
+		case map[string]any:
+			entries = []any{value}
+		case []any:
+			entries = value
+		default:
+			return "", false
+		}
+
+		return convertEntriesToTable(entries)
+	}
+}
+
+// Attempt to convert a slice of entries into a TSV table.
```

Comment:
"slice of entries" is a bit ambiguous.
You can just give the type `[]map[string]any` since this is an unexported function.